lib/memcached/memcached.rb

```diff
@@ -621,9 +621,9 @@ def reraise(key, ret)
 
   # Turn an array of keys into a hash of keys to servers.
   def inspect_keys(keys, server = nil)
-    Hash[*Array(keys).map do |key|
+    Hash[Array(keys).map do |key|
       [key, server || server_by_key(key)]
-    end.flatten]
+    end]
```

Comment:
If we just wanted a performance fix for large arrays of keys, then we could avoid the intermediate objects and add entries to the final hash directly.

```ruby
    ret = {}
    Array(keys).each do |key|
      ret[key] = server || server_by_key(key)
    end
    ret
```

If you wanted to go beyond the myopic view of performance problems with this private method and look at what we actually want for error message reraise uses this in.  It might make sense to include a maximum number of keys, with `...` to indicate that there are more.  Also, it looks like the message would be misleading when a `server` argument is applied, since it doesn't seem like that actually implies the same server was used for all the keys.

Reply:
> If you wanted to go beyond the myopic view of performance problems with this private method and look at what we actually want for error message reraise uses this in.

Yes! @arthurnn said the same thing too and I totally agree. 👍 Please feel free to co-opt this PR with a more holistic change or close it in favour of a different one. 🙇‍♂️